docs/tutorials/deadlock.rst

```diff
@@ -0,0 +1,92 @@
+Deadlock
+===============================
+
+Intro
+---------
+
+This lesson is meant to familiarize you with PyStack with a classical problem of locks acquisition.
+
+We will intentionally perform a deadlock, it a classic concurrency problem where improper ordering of lock acquisition can lead to deadlock,
+a situation where two or more threads are waiting for each other to release resources, causing the program to hang indefinitely.
+
+Development Environment Setup
+-----------------------------
+
+Navigate to the `PyStack GitHub repo <https://github.com/bloomberg/pystack>`_ and get a copy of the
+source code. You can either clone it, or just download the zip, whatever is your preference here.
+
+You will also need a terminal with ``python3`` installed, preferably one of the latest versions.
+
+Once you have the repo ready to navigate, ``cd`` into the docs/tutorials folder:
+
+.. code:: shell
+
+    cd docs/tutorials/
+
+It is here where we will be running the tests and exercises for the remainder of the tutorial.
+For reference here are the official `python3 venv docs <https://docs.python.org/3/library/venv.html>`_.
+You can also just follow along with the commands below.
+
+Let's go ahead and setup our virtual environment.
+
+.. code:: shell
+
+    python3 -m venv .venv
+
+Once your virtual environment has been created, you can activate it like so:
+
+.. code:: shell
+
+    source .venv/bin/activate
+
+You can confirm activation was successful if your terminal prompt is prefixed with ``(.venv)``.
+With our virtual environment ready, we can go ahead and install all the dependencies required
+for the tutorial.
+
+.. code:: shell
+
+    python3 -m pip install -r requirements-tutorial.txt
+
+Keep your virtual environment activated for the rest of the tutorial, and you should be able to run
+any of the commands in the exercises.
+
+PyStack remote use
+------------------------
+
+PyStack remote has the ability analyze the status of a running (remote) process.
+
+Executing the deadlock
+^^^^^^^^^^^^^^^^^^^^^^
+
+Assuming that we are in the `tutorials` directory, we can run the `deadlock script <https://github.com/bloomberg/docs/tutorials/deadlock.py>`_ with:
+
+.. code:: shell
+
+    python3 deadlock.py
+
+This script will intentionally perform a deadlock and the first message will contain the process ID.
+Another option is to find it with:
+
+.. code:: shell
+
+    ps | grep deadlock.py
+
+After the deadlock occurs we can use the PyStack command to analyze the process (replace the pid with the number of the process):
+
+.. code:: shell
+
+    pystack remote pid
```

Comment:
In order to make it a bit clearer what needs to be done, using `<PID>` makes the command fail immediately if it isn't replaced.

```suggestion
After the deadlock occurs we can use the PyStack command to analyze the
process (replace <PID> with the process ID from the previous step):

.. code:: shell

    pystack remote <PID>
```